### PR TITLE
Added bindings for datetime/date/time inputs

### PR DESCRIPTION
**Pull request by mereszeta**

### CODE REVIEW

File: core/.js/src/main/scala/io/udash/bindings/Bindings.scala [hunk 1]
```diff
@@ -34,9 +37,9 @@ trait Bindings {
     document.createTextNode("")
 
   /**
-    * Renders component with provided timeout.
-    * It's useful to render heavy components after displaying the main view.
-    */
+   * Renders component with provided timeout.
```
Comment (ddworak) [inline core/.js/src/main/scala/io/udash/bindings/Bindings.scala:40]: Please move the indentation changes to a separate PR.

File: core/.js/src/main/scala/io/udash/bindings/inputs/Input.scala [hunk 2]
```diff
@@ -56,3 +56,13 @@ object PasswordInput extends Input("password")
...
+
+/** Datetime input (Doesn't work on firefox - falls back to text input). (HTML5) */
+object DateTimepicker extends Input("datetime-local")
+
+/** Date input. (HTML5) */
+object Datepicker extends Input("date")
+
+/** Time input. (HTML5) */
+object Timepicker extends Input("time")
+
```
Comment (ddworak) [inline core/.js/src/main/scala/io/udash/bindings/inputs/Input.scala:68]: Any reason to rename those to "pickers" in such a low level API? They're still inputs according to docs.

**Review by ddworak — CHANGES_REQUESTED** [review]
Please remember to also add a demo to `io.udash.web.guide.views.frontend.FrontendFormsView`

File: core/.js/src/main/scala/io/udash/bindings/inputs/Input.scala (hunk 2 above)
Comment (mereszeta) [reply core/.js/src/main/scala/io/udash/bindings/inputs/Input.scala:68]: I don't understand I'm afraid

File: guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/demos/DateDemo.scala [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  private val ((firstInput, secondInput), source) = {
+    import io.udash._
+    import io.udash.bootstrap.form.UdashInputGroup
+    import io.udash.bootstrap.utils.BootstrapStyles._
+    import io.udash.css.CssView._
+    import scalatags.JsDom.all._
+    import org.scalajs.dom.html.Div
+    import scalatags.JsDom
+
+    val dateProperty = Property("")
```
Comment (ddworak) [inline guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/demos/DateDemo.scala:20]: nit: `Property.blank[String]`?

File: guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/demos/DateDemo.scala [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    def input: JsDom.TypedTag[Div] = div(Grid.row)(
+      div(Grid.col(4, ResponsiveBreakpoint.Medium))(
+        UdashInputGroup()(
+          UdashInputGroup.input(
+            DateInput(dateProperty)().render
+          ),
+        ).render
```
Comment (ddworak) [inline guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/demos/DateDemo.scala:28]: redundant render

File: guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/demos/DateTimeLocalDemo.scala [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import io.udash.web.guide.demos.AutoDemo
+import io.udash.web.guide.styles.partials.GuideStyles
+import io.udash._
+
+import scalatags.JsDom.all._
+
+
+object DateTimeLocalDemo extends AutoDemo {
+
+  private val dateTimeProperty = Property("")
```
Comment (ddworak) [inline guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/demos/DateTimeLocalDemo.scala:12]: maybe blank

File: guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/demos/DateTimeLocalDemo.scala [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    import org.scalajs.dom.html.Div
+    import scalatags.JsDom
+
+    def input: JsDom.TypedTag[Div] = div(Grid.row)(
+      div(Grid.col(4, ResponsiveBreakpoint.Medium))(
+        UdashInputGroup()(
+          UdashInputGroup.input(
+            DateTimeLocalInput(dateTimeProperty)().render
+          ),
+        ).render
```
Comment (ddworak) [inline guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/demos/DateTimeLocalDemo.scala:29]: redundant

File: guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/demos/TimeDemo.scala [hunk 7]
```diff
@@ -0,0 +1,49 @@
+package io.udash.web.guide.views.frontend.demos
+
+import io.udash.web.guide.styles.partials.GuideStyles
+import io.udash.web.guide.demos.AutoDemo
+import io.udash.web.guide.views.frontend.demos.DateDemo.{firstInput, secondInput}
+import scalatags.JsDom.all._
+
+object TimeDemo extends AutoDemo {
+
+
```
Comment (ddworak) [inline guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/demos/TimeDemo.scala:10]: nit: empty

File: guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/demos/TimeDemo.scala [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    val timeProperty = Property("")
+
+    def input: JsDom.TypedTag[Div] = div(Grid.row)(
+      div(Grid.col(4, ResponsiveBreakpoint.Medium))(
+        UdashInputGroup()(
+          UdashInputGroup.input(
+            TimeInput(timeProperty)().render
+          ),
+        ).render
```
Comment (ddworak) [inline guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/demos/TimeDemo.scala:28]: redundant

File: guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/demos/TimeDemo.scala [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+  override protected def demoWithSource(): (Modifier, Iterator[String]) = {
+    import io.udash.bootstrap.utils.BootstrapStyles._
+    import io.udash.css.CssView._
+    (div(id := "time-input-demo", GuideStyles.frame, GuideStyles.useBootstrap)(
+      form(containerFluid)(
+        firstInput, br, secondInput
+      )
+    ), source.linesIterator.take(source.linesIterator.size - 2))
+  }
+}
```
Comment (ddworak) [inline guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/demos/TimeDemo.scala:49]: Have you considered using an abstraction over these 3 demos? They're very similar in all aspects.

File: guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/demos/DateTimeLocalDemo.scala [hunk 10]
```diff
@@ -0,0 +1,50 @@
+package io.udash.web.guide.views.frontend.demos
+
+import io.udash.web.guide.demos.AutoDemo
+import io.udash.web.guide.styles.partials.GuideStyles
+import io.udash._
+
+import scalatags.JsDom.all._
+
+
+object DateTimeLocalDemo extends AutoDemo {
```
[the remark this answers is not in the document]
Comment (mereszeta) [reply guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/demos/DateTimeLocalDemo.scala:10]: We could, but this is a low level api example, I don't want to hide that here

File: guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/demos/TimeDemo.scala (hunk 9 above)
Comment (mereszeta) [reply guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/demos/TimeDemo.scala:49]: No, I thought that the example(due to source codes presented) should be as "standalone" as possible

File: guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/demos/DateTimeLocalDemo.scala (hunk 10 above)
[the remark this answers is not in the document]
Comment (ddworak) [reply guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/demos/DateTimeLocalDemo.scala:10]: Then you need a (visible) info in the demo.

[the remark this answers is not in the document]
Comment (mereszeta) [reply guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/demos/DateTimeLocalDemo.scala:10]: Yeah, I'll add the link to documentation with info that firefox does not support it

File: guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/FrontendFormsView.scala [hunk 11]
```diff
@@ -82,6 +93,15 @@ class FrontendFormsView extends View with CssView {
     p("Notice that the only difference is the type of the used property."),
     multiSelectSnippet,
     ForceBootstrap(multiSelectDemo),
+    h3("Date"),
+    p("Below examples show how to utilise `datetime-local`, `date` and `time` input bindings. All of them are bound to String property, " +
```
Comment (ddworak) [inline guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/FrontendFormsView.scala:97]: We don't really render backticks - you need to do that in tags here.

File: guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/FrontendFormsView.scala [hunk 12]
```diff
@@ -82,6 +93,15 @@ class FrontendFormsView extends View with CssView {
     p("Notice that the only difference is the type of the used property."),
     multiSelectSnippet,
     ForceBootstrap(multiSelectDemo),
+    h3("Date"),
```
Comment (ddworak) [inline guide/guide/.js/src/main/scala/io/udash/web/guide/views/frontend/FrontendFormsView.scala:96]: Date & time ?

**Review by ddworak — APPROVED** [review]
LGTM with minor comments
